src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -638,10 +638,21 @@ static double validateAndParseReal(Object input) {
     } else if (input instanceof Number) {
       return ((Number) input).doubleValue();
     } else if (input instanceof String) {
+      String stringInput = (String) input;
       try {
-        return Double.parseDouble((String) input);
+        return Double.parseDouble(stringInput);
       } catch (NumberFormatException err) {
-        throw valueFormatNotAllowedException(input, "REAL", "Not a valid decimal number");
+        stringInput = stringInput.toLowerCase();
+        switch (stringInput.toLowerCase()) {
```

Comment:
According to https://docs.snowflake.com/en/sql-reference/data-types-numeric.html#precision, the server side behavior is that everything needs to be in a single quote and are case sensitive, I think we should be following the same thing

Reply:
My bad, i thought it says case-sensitive